script/sync_prod_to_gcp/subscribe_submissions.py

```diff
@@ -0,0 +1,279 @@
+
+    Since upload() looks at the size of bucket object / CIT file to decide copy or not
+    copy, this will attempt to upload the versioned and latest at the same time but the uploading
+    may or may not happen.
+    """
+    try:
+        json_str = message.data.decode('utf-8')
+    except UnicodeDecodeError:
+        logger.error(f"bad data {str(message.message_id)}", extra=log_extra)
+        return ("", [])
+
+    try:
+        data = json.loads(json_str)
+    except Exception as _exc:
+        logger.warning(f"bad({message.message_id}): {json_str[:1024]}", extra=log_extra)
+        return ("", [])
+
+    publish_type = data.get('type') # cross | jref | new | rep | wdr
+    paper_id = data.get('paper_id')
+    version = data.get('version')
+    log_extra["arxiv_id"] = paper_id
+
+    # The source may not have a consistent .tar.gz name.
+    # Some other possibilities:
+    #
+    # {paperidv}.pdf pdf only submission
+    # {paperidv}.gz single file submisison
+    # {paperidv}.html.gz html source submission
+    #
+    # NOTE: This cannot handle this special case.
+    # I think there is one case in the system where there is a submission with two source files
+    # that the admins manually crafted. It would have: {paperidv}.pdf and {paperidv}.tar.gz
+    # This was to get a pdf only submission with ancillary files.
+
+    # Now, the submission file extension is provided in the message
+    src_ext = data.get('src_ext')
+    # Not sure I have to do this but better safe than sorry
+    src_ext = src_ext.strip() if src_ext else None
+    if src_ext:
+        submission_exts = [src_ext.strip(), ".abs"]
+    else:
+        src_ext = None
+        # cast a wider net for older queue element
+        submission_exts = [".pdf", ".gz", ".html.gz", ".tar.gz", ".abs"]
+
+    xid_latest = Identifier(f"{paper_id}")
+    logger.info("Processing %s.v%s:%s", xid_latest.ids, str(version), str(src_ext), extra=log_extra)
+    archive = ('arxiv' if not xid_latest.is_old_id else xid_latest.archive)
+    pairs = []
+
+    latest_dir = f"{FTP_PREFIX}{archive}/papers/{xid_latest.yymm}"
+    for dotext in submission_exts:
+        src_path = f"{latest_dir}/{xid_latest.id}{dotext}"
+        if os.path.exists(src_path):
+            pairs.append((src_path, path_to_bucket_key(src_path)))
+        else:
+            if src_ext is not None:
+                logger.error("Source does not exist: %s", src_path, extra=log_extra)
+
+    # if it is new/cross/jref just /data/ftp/arxiv/papers/{YYMM}/{paperidv}.* needs to be synced.
+    # If it is a replacement or wdr additionally /data/orig/arxiv/papers/{YYMM}/{paperid}{version-1}.* needs to be synced.
+    if publish_type in ["rep", "wdr"]:
+        prev_version = 1
+        try:
+            prev_version = max(1, int(version) - 1)
+        except:
+            pass
+        versioned_parent = f"{ORIG_PREFIX}{archive}/papers/{xid_latest.yymm}"
+        for dotext in submission_exts:
```

Comment:
I'm not totally sure submission_exts is going to be the full list here. If a src_ext is sent in a wdr message then the submission_exts list isn't going to be correct for the version-1 copy?